src/pages/Facility/settings/devices/components/ClampableText.tsx

```diff
@@ -0,0 +1,54 @@
+import { t } from "i18next";
+import { useEffect, useRef, useState } from "react";
+
+import { Button, ButtonVariant } from "@/components/ui/button";
+
+interface ClampableTextProps {
+  text: string;
+  linesToClamp?: number;
+  className?: string;
+  toggleTextBtnVariant?: ButtonVariant;
+  toggleTextBtnClassName?: string;
+}
+
+const ClampableText: React.FC<ClampableTextProps> = ({
+  text,
+  linesToClamp = 1,
+  className = "",
+  toggleTextBtnVariant = "link",
+  toggleTextBtnClassName = "px-0 hover:cursor-pointer text-primary-600",
+}) => {
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [hasOverflow, setHasOverflow] = useState(false);
+  const textRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    if (textRef.current) {
+      const isOverflowing =
+        textRef.current.scrollHeight > textRef.current.clientHeight;
+      setHasOverflow(isOverflowing);
+    }
+  }, [text, textRef]);
+
+  return (
+    <div className={`${className}`}>
+      <div
+        ref={textRef}
+        className={!isExpanded ? `line-clamp-${linesToClamp}` : ""}
+      >
```

Comment:
tailwind will not be able to compile this class!

https://tailwindcss.com/docs/detecting-classes-in-source-files#dynamic-class-names